Create an L2 listing policy (#5576)
ethereum/ethereum-org-website: src/content/contributing/adding-layer-2/index.md

---

> title: Adding Layer 2's

Reviewer: Grammar nitpick, should it be "Layer 2's"? Or "Layer 2s"? What do you say @minimalsm 😄?

description: The policy we use when adding layer 2's to ethereum.org
lang: en
sidebar: true
---

> # Adding layer 2's {#adding-layer-2}

Reviewer 2: ```suggestion
# Adding layer 2s {#adding-layer-2}
```

Anyone is free to suggest new layer 2's on ethereum.org.

We currently list them on:

> - [ethereum.org/layer-2](/layer-2/)

Reviewer: We shouldn't add this page until it exists. Currently we list L2s on:
- https://ethereum.org/en/developers/docs/scaling/optimistic-rollups/
- https://ethereum.org/en/developers/docs/scaling/zk-rollups/
- ...any others?

Author: For sure, wasn't quite sure when the plan was to put this out, but ill update this to current pages and we can come back after layer-2 page is out.

This page is a beginner/intermediate level page on layer 2's, and on this page a user is able to see some prominent layer 2 projects that they can easily get onboarded on.

> Because of this context, we need some specific information when you suggest a layer 2.

Reviewer: I think we should ask for the same base level of information about any L2 project, regardless of where we put it (e.g. the beginner's page vs. the developer docs). Perhaps we ask for additional info for the beginner's page but we can start with a base requirement for any L2 listing.

> ## What we need {#what-we-need}

Reviewer: I'd say let's keep the same structure of existing listing criteria pages, e.g. "[How we decide](https://ethereum.org/en/contributing/adding-developer-tools/#ways-to-contribute)" or "[The decision framework](https://ethereum.org/en/contributing/adding-products/#decision-framework)".

> Some information we will need beyond standard information of the layer 2 are:

Reviewer: "standard information of the layer 2" is pretty vague, what are you thinking here? Perhaps we say in addition to the information we request from a [product](https://ethereum.org/en/contributing/adding-products/) or [developer tool](https://ethereum.org/en/contributing/adding-developer-tools/)?

Reviewer 2: To avoid confusion let's be explicit on this page **exactly** what information we require instead of expecting users to combine this page with some information from the product/developer tools.

Reviewer: Agreed 💯

> - [L2Beat page](https://l2beat.com) - in order to be considered we need this project to be featured on L2beat. This resource provides a great risk assessment of layer 2 projects. If the project is not featured on L2Beat, they will not be considered for ethereum.org.

Reviewer: ```suggestion
- [Listing on L2Beat](https://l2beat.com) - in order to be considered we need this project to be featured on L2beat. This resource provides a great risk assessment of layer 2 projects. If the project is not featured on L2Beat, they will not be considered for ethereum.org.
```
L2BEAT should be all caps.

- Is the project open source?
- What kind of layer 2 is it? Optimistic? ZK? Validium?
- Bridges - how are users able to onboard to the layer 2?

> - Exchange support - are users able to deposit directly from an exchange?

Reviewer 2: Wallet support too?

- How long has the layer 2 been live?
- Links to dapps in the layer 2 ecosystem - we want to be able to provide information on what users can expect to be able to do on these layer 2's.
- Token contract lists - since assets will have a new address on layer 2, we would like to be able to help users onboard with trusted resources for token contracts on the layer 2.
- Is the project a fork?

> - Is the project audited?

Reviewer 2: Is this an external security audit?

Reviewer: Agree we should be more explicit about what this means.

We need this information so that we can provide the best onboarding experience for users learning about layer 2's.

## Some additional things we may look for {#some-additional-things-we-may-look-for}

> TVL of the layer 2. We want to see sustained adoption of the network, so if it is only being used short term for things like yield farming and then liquidity leaves, we may take this into consideration when choosing to accept a project or not.

Reviewer: I don't see a reason to separate this into a separate section. I think this can be part of the larger list of e.g. "How we decide".

> Is the project being actively maintained. We don't want to list layer 2's that don't have an active team supporting the project.

Reviewer: Same, I think this can be part of the list above.

Reviewer 2: I think we should be explicit about being under active development (we probably aren't likely to say any L2 is a finished product for a long time).

## Add your layer 2 {#add-exchange}

If you want to add a layer 2 to ethereum.org, create an issue on GitHub.

> <ButtonLink to="https://github.com/ethereum/ethereum-org-website/issues/new/choose">

Reviewer 2: This links directly to the issue template above.

```suggestion
<ButtonLink to="https://github.com/ethereum/ethereum-org-website/issues/new?&template=suggest_layer2.md">
```

- [Zk rollups](/developers/docs/scaling/zk-rollups/)
<!-- TODO: Add layer-2 page -->

## How we decide {#how-we-decide}

Layer 2 submissions will be assessed by the following criteria:

**[Listing on L2BEAT](https://l2beat.com)**

- In order to be considered we need this project to be featured on L2BEAT. This resource provides a great risk assessment of layer 2 projects. **If the project is not featured on L2BEAT, they will not be considered for ethereum.org.**

**Is the project open source?**

**What kind of layer 2 is it?**

- Optimistic
- ZK
- State channel

**Bridges**

- How are users able to onboard to the layer 2?

**Exchange support**

- Are users able to deposit directly from an exchange?

**Links to dapps in the layer 2 ecosystem**

- We want to be able to provide information on what users can expect to be able to do on this layer 2.

**Token contract lists**

- Since assets will have a new address on layer 2, if there is a token list resource available please share.

**How long has the layer 2 been live?**

**Is the project a fork?**

- If the layer 2 is a fork, what project is it a fork of?

**What wallets can users use to access this layer 2?**

> - Can users just import an RPC? What is that RPC?

Reviewer: Not sure what we're getting at here. Are we asking if there's a public RPC people can use to interact with this network? e.g. `https://arb1.arbitrum.io/rpc` for Arbitrum?

Author: Yes, will make this more clear.

- What wallets natively support this layer 2?

**Has the project been externally audited?**

- Include links to report and/or repo.

> **TVL history of the layer 2**

Reviewer: I might update this to something similar as we have on developer tools listing:

```
**Is the tool widely used?**

- We will consider metrics such as GitHub stars, download statistics, and whether it is used by known companies or projects
```

So maybe something like

```
**Is the layer 2 widely used?**

- We will consider metrics such as TVL history, transaction statistics, and whether it is used by known companies or projects
```

---
title: Adding Layer 2s
description: The policy we use when adding a layer 2 to ethereum.org
lang: en
sidebar: true
---

# Adding layer 2s {#adding-layer-2}

We want to make sure we list the best resources possible so users can navigate the layer 2 space in a safe and confident manner.

Anyone is free to suggest adding a layer 2 on ethereum.org. If there's a layer 2 that we have missed, **please suggest it!**

We currently list L2s on the following pages:

- [Optimistic rollups](/developers/docs/scaling/optimistic-rollups/)
- [Zero-knowledge rollups](/developers/docs/scaling/zk-rollups/)
<!-- TODO: Add layer-2 page -->

## How we decide {#how-we-decide}

Layer 2 submissions will be assessed by the following criteria:

**[Listing on L2BEAT](https://l2beat.com)**

> - In order to be considered, this project to be featured on L2BEAT. This resource provides a great risk assessment of layer 2 projects. **If the project is not featured on L2BEAT, they will not be considered for ethereum.org.**

Reviewer 2: Feel free to edit but the unasked question I'm trying to answer is: why would ethereum.org trust L2BEAT's analysis instead of doing it themselves?

```suggestion
- In order to be considered, this project must be featured on L2BEAT. L2BEAT provides a great risk assessment of layer 2 projects that we lean on for evaluating L2 projects. **If the project is not featured on L2BEAT, we will not list them as an L2 on ethereum.org.**
```

[Learn how to add your L2 project to L2BEAT](https://github.com/l2beat/l2beat/blob/master/CONTRIBUTING.md).

**Is the project open source?**

> **What kind of layer 2 is it?**

Reviewer 2: ```suggestion
**What category of layer 2 is it?**
```

- Optimistic rollup
- Zero-knowledge rollup
- State channel

> **Bridges**

Reviewer 2: ```suggestion
**Which bridges are used to onboard?**
```

- How are users able to onboard to the layer 2?

> **Exchange support**

Reviewer 2: ```suggestion
**Which exchanges offer support for the project?**
```

- Are users able to deposit and/or withdraw directly from an exchange? If yes, which exchanges?

**Links to dapps in the layer 2 ecosystem**

> - We want to be able to provide information on what users can expect to be able to do on this layer 2.

Reviewer 2: Maybe we list Arbitrum's page here as an example here?

**Token contract lists**

- Since assets will have a new address on layer 2, if there is a token list resource available please share.

> **How long has the layer 2 been live?**
> 
> **Is the project a fork?**

Reviewer 2: Current wording makes us math. Also, we should ask if they can provide proof of launch dates.

```suggestion
**What date did the layer 2 go live?**

- If possible, please provide proof to substantiate the launch date.

**Is the project a fork?**
```

- If the layer 2 is a fork, what project is it a fork of?

**What wallets can users use to access this layer 2?**

- What is the public RPC information for the network? (RPC URL, Chain ID, Currency Symbol, Block Explorer URL)
- What wallets natively support this layer 2?

> **Has the project been externally audited?**

Reviewer 2: Any other security considerations we should ask for? Internal security team?

- Include links to report and/or repo.

**Is the layer 2 widely used?**

> - We will consider metrics such as TVL history, transaction statistics, and whether it is used by known companies or projects

Reviewer 2: Seems a little hazy. Perhaps phrasing it in the form of the questions we would ask when evaluating an L2 might be helpful.

e.g.

- Has the project maintained its users over time?

**Is the project being actively maintained?**

> - We won't list layer 2s that don't have an active team supporting the project.

Reviewer 2: ```suggestion
- We won't list a layer 2 that doesn't have an active team working on project.
```

> **Is there a block explorer?**

Reviewer 2: Adding this in case someone answers 'yes' without providing a link 😅

```suggestion
**Is there a block explorer?**

- Please provide a link to the block explorer.
```

Anyone is free to suggest adding a layer 2 on ethereum.org. If there's a layer 2 that we have missed, **please suggest it!**

We currently list L2s on the following pages:

- [Optimistic rollups](/developers/docs/scaling/optimistic-rollups/)
- [Zero-knowledge rollups](/developers/docs/scaling/zk-rollups/)
<!-- TODO: Add layer-2 page -->

Layer 2 is a relatively new and exciting paradigm for Ethereum. We've tried to create a fair framework for consideration on ethereum.org but the listing criteria will change and evolve over time.

## Information we require {#information-we-require}

Layer 2 submissions will be assessed by the following criteria:

**[Listing on L2BEAT](https://l2beat.com)**

- In order to be considered, this project must be featured on L2BEAT. L2BEAT provides a great risk assessment of layer 2 projects that we lean on for evaluating L2 projects. **If the project is not featured on L2BEAT, we will not list them as an L2 on ethereum.org.**

[Learn how to add your L2 project to L2BEAT](https://github.com/l2beat/l2beat/blob/master/CONTRIBUTING.md).

**Is the project open source?**

**What category of layer 2 is it?**

- Optimistic rollup
- Zero-knowledge rollup

> - State channel

Reviewer 3: Should we explicitly call out things like validiums/volitions/plasmas here as currently not being considered for listing? Could help prevent confusion

Author: Hmm, I think this should be fine for now since the options are listed, but maybe we see how this goes and reassess?

Reviewer 2: I agree @corwintines

**Which bridges are used to onboard?**

- How are users able to onboard to the layer 2?

**Which exchanges offer support for the project?**

- Are users able to deposit and/or withdraw directly from an exchange? If yes, which exchanges?

**Links to dapps in the layer 2 ecosystem**

- We want to be able to provide information on what users can expect to be able to do on this layer 2. (ex: https://portal.arbitrum.one/, https://www.optimism.io/apps/all)

**Token contract lists**

- Since assets will have a new address on layer 2, if there is a token list resource available please share.

**How long has the layer 2 been live?**

**Is the project a fork?**

- If the layer 2 is a fork, which project was forked?

**What wallets can users use to access this layer 2?**

- What is the public RPC information for the network? (RPC URL, Chain ID, Currency Symbol, Block Explorer URL)
- What wallets natively support this layer 2?

> **Has this project undergone an external security audit?**
> 
> - Include links to report and/or repo.

Reviewer 3: Should we mention bug bounties too?
```suggestion
**Has this project undergone an external security audit or bug bounty?**

- Include links to any reports and/or repos.
```

**Has the project maintained its users over time?**

- We will consider metrics such as TVL history, transaction statistics, and whether it is used by known companies or projects

**Is the project being actively maintained?**

- We won't list a layer 2 that doesn't have an active team working on project.

**Is there a block explorer?**

- Please provide a link to the block explorer.

> 

Reviewer 2: ```suggestion

### Other criteria: the nice-to-haves {#other-criteria-the nice-to-haves}

```

<!-- TODO: Add layer-2 page -->

Layer 2 is a relatively new and exciting paradigm for Ethereum. We've tried to create a fair framework for consideration on ethereum.org but the listing criteria will change and evolve over time.

## The decision framework {#decision-framework}

### Criteria for inclusion: the must-haves {#criteria-for-inclusion-the-must-haves}

**[Listing on L2BEAT](https://l2beat.com)**

- In order to be considered, this project must be listed on L2BEAT. L2BEAT provides a robust risk assessment of layer 2 projects that we lean on for evaluating L2 projects. **If the project is not featured on L2BEAT, we will not list them as an L2 on ethereum.org.**

[Learn how to add your L2 project to L2BEAT](https://github.com/l2beat/l2beat/blob/master/CONTRIBUTING.md).

**Open source**

**Layer 2 category**

We currently consider the following to be layer 2 solutions:

- Optimistic rollup
- Zero-knowledge rollup
- State channel
  _We do not consider other scaling solutions to be layer 2._

**Bridges**

- How are users able to onboard to the layer 2?

**Date project went live**

- Newer projects that have not been battle-tested by users are less likely to be listed.

> **Wallet support**
> 
> - What is the public RPC information for the network? (RPC URL, Chain ID, Currency Symbol, Block Explorer URL)
> - What wallets natively support this layer 2?

Reviewer 2: I think this is confusing.

- We can probably remove the first point and keep it in the issue template because any L2 will have a public RPC (Correct me if I'm wrong here)
- The second point is a nice to have

I think we can edit this and move it into the nice to haves

```suggestion
**Native wallet support**

- Do any wallets support the L2 natively?
```

## The decision framework {#decision-framework}

### Criteria for inclusion: the must-haves {#criteria-for-inclusion-the-must-haves}

**Listing on L2BEAT**

- In order to be considered, this project must be listed on [L2BEAT](https://l2beat.com). L2BEAT provides a robust risk assessment of layer 2 projects that we lean on for evaluating L2 projects. **If the project is not featured on L2BEAT, we will not list them as an L2 on ethereum.org.**
- [Learn how to add your L2 project to L2BEAT](https://github.com/l2beat/l2beat/blob/master/CONTRIBUTING.md).

**Open source**

- Your code must be accessible and you should accept PRs from the wider community.

**Layer 2 category**

We currently consider the following to be layer 2 solutions:

- Optimistic rollup
- Zero-knowledge rollup
- State channel

_We do not consider other scaling solutions to be layer 2._

**Bridges**

- How are users able to onboard to the layer 2?

**Date project went live**

- Newer projects that have not been battle-tested by users are less likely to be listed.

> **Wallet and tooling support**
> 
> - What is the public RPC information for the network? (RPC URL, Chain ID, Currency Symbol, Block Explorer URL)
> 

Reviewer 2: ```suggestion
```
